content/en/blog/_posts/2023-07-27-job-update-post.md

```diff
@@ -0,0 +1,207 @@
+Pods might produce undesired scale ups.
+
+### How can I use it
+
+This is an alpha feature, which you can enable by enabling the `JobPodReplacementPolicy`
+[feature gate](/docs/reference/command-line-tools-reference/feature-gates/) in
+your cluster.
+
+Once the feature is enabled you can use it by creating a new Job, which specifies
+`podReplacementPolicy` field as shown here:
+
+```yaml
+kind: Job
+metadata:
+  name: new
+  ...
+spec:
+  podReplacementPolicy: Failed
+  ...
+```
+
+Additionally, you can inspect the `.status.terminating` field of a Job. The value
+of the field is the number of Pods owned by the Job that are currently terminating.
+
+```shell
+kubectl get jobs/myjob -o yaml
+```
+
+```yaml
+apiVersion: batch/v1
+kind: Job
+status:
+  terminating: 3 # three Pods are terminating and have not yet reached the Failed phase
+```
+
+This can be particularly useful for external queueing controllers, such as
+[Kueue](https://github.com/kubernetes-sigs/kueue), that would calculate the
+quota and gate the start of a new Job until the resources are reclaimed from
+the currently terminating Job.
+
+### How can I learn more?
+
+- Read the KEP: [PodReplacementPolicy](https://github.com/kubernetes/enhancements/tree/master/keps/sig-apps/3939-allow-replacement-when-fully-terminated)
+
+## Job Backoff Limit per Index
+
+### What problem does this solve?
+
+Currently, for an [Indexed](/docs/concepts/workloads/controllers/job/#completion-mode)
+Jobs failures are counted towards the global limit of retries. This means, that
+if there is a consistently failing index, it would be restarted repeatedly until
+it exhausts the global limit of retries. Once the limit is exceeded the entire
+Job is marked failed and some indexes may
+
+This is problematic for use cases where you want to handle pod failures for
+every index independently. For example, if you are running integration tests
+where each index corresponds to a testing suite. In that case, you may want to
+account for possible flake tests allowing for 1 or 2 retries per suite.
+Additionally, when there are buggy suites, making some of the indexes failing
+consistently, you would like to terminate retries for that indexes, yet allowing
+other suites to complete.
+
+Thus, the feature allows you to:
+* complete execution of all indexes, despite some indexes failing,
+* better utilize the computational resources by avoiding unnecessary retries of consistently failing indexes.
+
+### How to use it?
+
+This is an alpha feature, which you can enable by enabling the
+`JobBackoffLimitPerIndex`
+[feature gate](/docs/reference/command-line-tools-reference/feature-gates/)
+in your cluster.
+
+Once the feature is enabled, you can create an Indexed Job with the
+`.spec.backoffLimitPerIndex` field specified.
+
+#### Example
+
+The following example demonstrates how to use this feature to make sure the
+Job executes all indexes, and the number of failures is controller per index.
+
+```yaml
```

Comment:
In the `PodReplacementPolicy` section we use an abbr. form of the spec.  Do you think we should expand those so that the examples are valid Job specs?

Reply:
Good question. I was also thinking about this, but decided to leave it as is for now.

Reply:
Sounds good.  I’m not sure if I should update mine or not.  I’ll leave this comment open for that reason.

Reply:
You don't need to put a full manifest in a blog article if that's not helpful.